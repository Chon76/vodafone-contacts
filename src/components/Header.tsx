import { Center, Divider, Heading } from 'native-base';

interface HeaderProps {
  title: string;
}

export default function Header({ title }: HeaderProps) {
  return (
    <Center>
      <Heading fontSize={'4xl'} pb={2} textAlign={'center'}>
        {title}
      </Heading>

      <Divider w={'10%'} h={1} bg={'primary'} rounded={'2xl'} />
    </Center>
  );
}
