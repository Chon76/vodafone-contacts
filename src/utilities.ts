import { Contact } from './types/Contact';

// eslint-disable-next-line import/prefer-default-export
export function getInitialLeters(contacts: Contact[]): Set<string> {
  const initialLetters: Set<string> = new Set();

  contacts.forEach((contact) => {
    initialLetters.add(contact.name.first[0].toUpperCase());
    initialLetters.add(contact.name.last[0].toUpperCase());
  });

  console.log('initials', initialLetters);

  return initialLetters;
}
