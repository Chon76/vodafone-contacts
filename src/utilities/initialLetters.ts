import { Contact } from '../types/Contact';

/**
 *
 * @param contacts an array of {@link Contact}
 * @returns the initial letters of the names of the contact
 */
// eslint-disable-next-line import/prefer-default-export
export function getInitialLetters(contacts: Contact[]): Set<string> {
  const initialLetters: Set<string> = new Set();

  contacts.forEach((contact) => {
    initialLetters.add(contact.name.first[0].toUpperCase());
    initialLetters.add(contact.name.last[0].toUpperCase());
  });

  return initialLetters;
}
