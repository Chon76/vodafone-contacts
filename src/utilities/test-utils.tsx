/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderOptions } from '@testing-library/react-native';
import React, { ReactElement } from 'react';
import { NativeBaseProvider } from 'native-base';
import { NavigationContainer } from '@react-navigation/native';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

const wrapper = ({ children }: { children: React.ReactNode }) => {
  return (
    <NativeBaseProvider initialWindowMetrics={inset}>
      <NavigationContainer>{children}</NavigationContainer>
    </NativeBaseProvider>
  );
};

// wrap tests with necessery providers
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper, ...options });

export * from '@react-navigation/native';
export { customRender as render };
