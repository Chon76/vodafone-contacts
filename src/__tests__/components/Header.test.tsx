import Header from '../../components/Header';

import { render } from '../../utilities/test-utils';

describe('header', () => {
  it('renders correctly', () => {
    const fab = render(<Header title="Test title" />);

    expect(fab.toJSON()).toMatchSnapshot();
  });
});
