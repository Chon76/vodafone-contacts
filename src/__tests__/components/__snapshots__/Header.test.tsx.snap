// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <Text
      dataSet={{}}
      style={
        {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 36,
          "fontStyle": "normal",
          "fontWeight": "700",
          "letterSpacing": 0,
          "lineHeight": 45,
          "paddingBottom": 8,
          "textAlign": "center",
          "textDecorationLine": undefined,
        }
      }
    >
      Test title
    </Text>
    <View
      aria-orientation="horizontal"
      dataSet={{}}
      style={
        {
          "backgroundColor": {
            "100": "#cffafe",
            "200": "#a5f3fc",
            "300": "#67e8f9",
            "400": "#22d3ee",
            "50": "#ecfeff",
            "500": "#06b6d4",
            "600": "#0891b2",
            "700": "#0e7490",
            "800": "#155e75",
            "900": "#164e63",
          },
          "borderRadius": 16,
          "height": 4,
          "width": "10%",
        }
      }
      thickness="1"
    />
  </View>
</RNCSafeAreaProvider>
`;
